### PR TITLE
fix: use tryadd instead of containskey guard clause

### PR DESCRIPTION
**Pull request by IXLLEGACYIXL**
# PR Details

Uses the new NET8 TryAdd instead of ContainsKey  guard clause + Add
It's faster as you dont need a double lookup in the dictionary

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] **I have built and run the editor to try this change out.**

### CODE REVIEW

**Review by Jklawreszuk — APPROVED** [review]
Looks legit to me

File: sources/engine/Stride.Physics/Engine/PhysicsComponent.cs [hunk 1]
```diff
@@ -760,10 +760,8 @@ public void IgnoreCollisionWith(PhysicsComponent other, CollisionState state)
                 {
                     if(ignoreCollisionBuffer == null)
                         ignoreCollisionBuffer = new Dictionary<PhysicsComponent, CollisionState>();
```
Comment (Kryptos-FR) [inline sources/engine/Stride.Physics/Engine/PhysicsComponent.cs:762]: While we are at it, we can simplify these lines.

```csharp
ignoreCollisionBuffer ??= new()
```

File: sources/engine/Stride.Physics/Engine/PhysicsComponent.cs [hunk 2]
```diff
@@ -760,10 +760,8 @@ public void IgnoreCollisionWith(PhysicsComponent other, CollisionState state)
...
-                    if(ignoreCollisionBuffer.ContainsKey(other))
+                    if(!ignoreCollisionBuffer.TryAdd(other, state))
                         ignoreCollisionBuffer[other] = state;
```
[the remark this answers is not in the document]
Comment (Kryptos-FR) [reply sources/engine/Stride.Physics/Engine/PhysicsComponent.cs:764]: We could keep `TryAdd` to be consistent with the other changes.

File: sources/tools/Stride.TextureConverter/Backend/Wrappers/FINetWrapper/FreeImageWrapper.cs [hunk 3]
```diff
@@ -4323,15 +4323,12 @@ public static bool SetImageComment(FIBITMAP dib, string comment)
 				return result;
 			}
 			tag = new MetadataTag(_tag, model);
-			if (metaDataSearchHandler.ContainsKey(result))
+			if (!metaDataSearchHandler.TryAdd(result, model))
 			{
 				metaDataSearchHandler[result] = model;
 			}
```
[the remark this answers is not in the document]
Comment (Kryptos-FR) [reply sources/tools/Stride.TextureConverter/Backend/Wrappers/FINetWrapper/FreeImageWrapper.cs:4329]: We could keep `TryAdd` to be consistent with the other changes.

File: sources/engine/Stride.Physics/Engine/PhysicsComponent.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Eideren) [reply sources/engine/Stride.Physics/Engine/PhysicsComponent.cs:764]: What do you mean ? Replace those two lines with just a `TryAdd` ? `TryAdd` won't assign the value if the key already exists, which is important for this specific use case

File: sources/tools/Stride.TextureConverter/Backend/Wrappers/FINetWrapper/FreeImageWrapper.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (IXLLEGACYIXL) [reply sources/tools/Stride.TextureConverter/Backend/Wrappers/FINetWrapper/FreeImageWrapper.cs]: [1de63d9](https://github.com/stride3d/stride/pull/2299/commits/1de63d967403857a67ee21de981b86ec83ed3331)
done

File: sources/engine/Stride.Physics/Engine/PhysicsComponent.cs (hunk 1 above)
Comment (IXLLEGACYIXL) [reply sources/engine/Stride.Physics/Engine/PhysicsComponent.cs]: [8713632](https://github.com/stride3d/stride/pull/2299/commits/8713632f65e137ba1d048f48aa794e8f648107fb)
done

Comment (IXLLEGACYIXL) [reply sources/engine/Stride.Physics/Engine/PhysicsComponent.cs]: new() can be reduced to [] ... should i do that instead?

Comment (Kryptos-FR) [reply sources/engine/Stride.Physics/Engine/PhysicsComponent.cs]: Yes, it's even better.

File: sources/engine/Stride.Physics/Engine/PhysicsComponent.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kryptos-FR) [reply sources/engine/Stride.Physics/Engine/PhysicsComponent.cs]: @Eideren you're right. I misread it.

**Comment by IXLLEGACYIXL** [comment]
did all the review changes, i have everything i wanted for the pr

**Review by Kryptos-FR — APPROVED** [review]
LGTM

**Comment by Eideren** [comment]
Thanks !

**Pull request merged**